mono/utils/mono-dl.c

```diff
@@ -220,7 +228,7 @@ mono_dl_open (const char *name, int flags, char **error_msg)
 			if (error_msg)
 				*error_msg = NULL;
 			
-			lib = handler->load_func (name, lflags, error_msg, handler->user_data);
+			lib = handler->load_func (name, mono_dl_flags_external_to_internal (lflags), error_msg, handler->user_data);
```

Comment:
This doesn't seem right.  `lflags` is the result of `mono_dl_convert_flags` - so it's `RTLD_*`.  But `mono_dl_flags_external_to_internal` is masking on `MONO_DL_*`

Reply:
Oh, yep, good catch. Went too quickly on this.